### DISCUSSION TITLE
v2 Config: Allow outputs to be specified with inputs

### DISCUSSION THREAD

**Discussion by bhollis** [bufbuild/buf#3045] [Ideas]
The v2 config allows me to specify inputs in `buf.gen.yaml`, which would be great if all of my code was meant to be generated to the same place. Instead, what I want is to use the same plugins, the same *relative* output paths, but different output directories per input. If we could specify outputs along with inputs, we could get down to a single `buf generate` invocation - instead, we still have to call it once per input.

**Comment by doriable** [comment]
This is pretty commonly requested, we have an issue centralising and tracking the discussions on this https://github.com/bufbuild/buf/issues/3060, we will definitely be discussing this internally! Thanks for putting up this discussion, it helps us prioritise and gather these concerns :)
